### PR TITLE
submit jobspec as JSON

### PR DESCRIPTION
**Pull request by garlick**
This is a #1809, updated
* to use PyYAML instead of ruamel.yaml (I _think_ this is what we decided)
* pass black formatting
* fix travis Dockerfile to restore apt/yum cache prior to installing python-yaml

Whether this is the right direction may be an open question - #1909 was opened to track that decision.

(I still might need a bit of help understanding the proper procedure for submitting a PR with new package dependencies - though this seems to have worked for getting the initial PR through travis).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ lua-posix 	| lua-posix             | 			| *1*
 python-devel	| python-dev		| >= 2.7		|
 python-cffi	| python-cffi		| >= 1.1		|
 python-six	| python-six		| >= 1.9		|
+python-yaml	| python-yaml		|			|
```
Comment (SteVwonder) [inline README.md]: Based on the configure check, I think `>= 3.10.0` should be added to the third column.

**Comment by garlick** [comment]
Thanks - rebased on current master and made those changes (squashing the doc one).

**Comment by SteVwonder** [comment]
LGTM!

**Comment by garlick** [comment]
Thanks!

I'm done working on this if someone wants to press the button.

**Comment by grondo** [comment]
> I assume that a jobspec will be stored as a JSON in KVS. How should flux-sched's resource matching service construct a jobspec out of the JSON?

@dongahn, I think JSON is valid YAML and therefore libjobspec should be able to parse as JSON. We should give it a try though.

**Comment by grondo** [comment]
Small typo in commit message for a396cb5: `s/supplised/supplied/`

**Comment by dongahn** [comment]
@grondo: Ah good point. We should indeed try and test this.

**Comment by garlick** [comment]
> We should indeed try and test this.

`t2200-job-ingest.t` runs all the valid and invalid jobspec test input through the ingest module as JSON.  The ingest module still validates jobspec using the C++ libjobspec, so I think the testing of this is already covered in this PR.

**Comment by garlick** [comment]
> Small typo in commit message for a396cb5: s/supplised/supplied/

Thanks!  Fixed and re-pushed.

**Comment by grondo** [comment]
Ok, LGTM!

**Comment by garlick** [comment]
Thanks!

**Pull request merged**